packages/core/src/oidc/utils.ts

```diff
@@ -8,3 +8,14 @@ export const buildOidcClientMetadata = (metadata?: OidcClientMetadata): OidcClie
   postLogoutRedirectUris: [],
   ...metadata,
 });
+
+export const allowDomain = (origin: string, clientProperties: Record<string, unknown>) => {
+  if (!(CustomClientMetadataType.allowedDomains in clientProperties)) {
+    return false;
+  }
+
+  const allowedDomains = clientProperties[CustomClientMetadataType.allowedDomains] as string[];
+  const { host } = new URL(origin);
+
+  return allowedDomains.some((allowedDomain) => host.startsWith(allowedDomain));
```

Comment:
if use `startWith` here, all port numbers under the allowedDomain will be treated as a legal origin, is that what we expected?

Reply:
e.g. if developers want to check the port, it will change `http://localhost` to `http://localhost:3000`. `startWith` works